workspaces/ui-v2/src/optic-components/pages/diffs/ReviewEndpointDiffPage/ReviewEndpointDiffPageContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { FC, useContext } from 'react';
+import { RouteComponentProps } from 'react-router';
+import { IForkableSpectacle } from '@useoptic/spectacle';
+
+import { useEndpointDiffs } from '../../../hooks/diffs/useEndpointDiffs';
+import { useShapeDiffInterpretations } from '../../../hooks/diffs/useDiffInterpretations';
+import { useSharedDiffContext } from '../../../hooks/diffs/SharedDiffContext';
+import { useEndpoint } from '../../../hooks/useEndpointsHook';
+import { SpectacleContext } from '../../../../spectacle-implementations/spectacle-provider';
+
+import { ReviewEndpointDiffPage } from './ReviewEndpointDiffPage';
+
+export const ReviewEndpointDiffContainer: FC<
```

Comment:
I split this page component into multiple components for a couple of reasons.

We want to set initial state on the initial render for the initial contribution. If the endpoint is `null` - i.e. still loading, we don't actually inject the old contribution.

As a result, I updated this to have a loading state + shard this into multiple pages in an attempt to:
- handle typing and loading states more upfront (so that we don't silently swallow errors and force us to have stronger data constraints)
- Hopefully split up concerns so that this component is easier to digest what it does